src/platform_impl/apple/appkit/view.rs

```diff
@@ -1034,42 +1047,63 @@ impl WinitView {
     }
 
     fn mouse_click(&self, event: &NSEvent, button_state: ElementState) {
-        let button = mouse_button(event);
-
         self.update_modifiers(event, false);
 
-        self.queue_event(WindowEvent::MouseInput {
+        let view_point = self.mouse_view_point(event);
+
+        if !self.is_in_window(view_point) {
+            return;
+        }
+
+        let button = mouse_button(event);
+
+        self.queue_event(WindowEvent::PointerButton {
             device_id: DEVICE_ID,
             state: button_state,
-            button,
+            position: view_point.to_physical(self.scale_factor()),
+            button: button.into(),
         });
     }
 
     fn mouse_motion(&self, event: &NSEvent) {
+        self.update_modifiers(event, false);
+
+        let view_point = self.mouse_view_point(event);
+
+        if !self.is_in_window(view_point) {
+            return;
+        }
+
+        self.queue_event(WindowEvent::PointerMoved {
+            device_id: DEVICE_ID,
+            position: view_point.to_physical(self.scale_factor()),
+            source: PointerSource::Mouse,
+        });
+    }
+
+    fn mouse_view_point(&self, event: &NSEvent) -> LogicalPosition<f64> {
         let window_point = unsafe { event.locationInWindow() };
         let view_point = self.convertPoint_fromView(window_point, None);
+
+        LogicalPosition::new(view_point.x, view_point.y)
+    }
+
+    fn is_in_window(&self, position: LogicalPosition<f64>) -> bool {
```

Comment:
Nit: Use AppKit terminology here.

```suggestion
    fn is_in_frame(&self, position: LogicalPosition<f64>) -> bool {
```

Reply:
So I did figure that this check didn't make sense in `mouse_click()` as well, so I removed this function entirely in https://github.com/rust-windowing/winit/pull/3876/commits/2aed0efa61b37424b481e7cc0609b3e35e7fd2ff.
Apologies for all the code sorting here, I just reverted back to the exact order it was before this PR so the squashed diff becomes as small as possible.

Let me know if I got anything wrong here!

Reply:
Looks fine, thanks!